flink-filesystems/flink-hadoop-fs/src/main/java/org/apache/flink/runtime/util/HadoopUtils.java

```diff
@@ -112,11 +112,12 @@ public static Configuration getHadoopConfiguration(org.apache.flink.configuratio
 		return result;
 	}
 
-	public static boolean isCredentialsConfigured(UserGroupInformation ugi, boolean useTicketCache) throws Exception {
-		if (UserGroupInformation.isSecurityEnabled()) {
+	public static boolean isKerberosCredentialsConfigured(UserGroupInformation ugi, boolean useTicketCache) {
+		if (UserGroupInformation.isSecurityEnabled()
+			&& ugi.getAuthenticationMethod() == UserGroupInformation.AuthenticationMethod.KERBEROS) {
 			if (useTicketCache && !ugi.hasKerberosCredentials()) {
 				// a delegation token is an adequate substitute in most cases
-				if (!HadoopUtils.hasHDFSDelegationToken()) {
+				if (!HadoopUtils.hasHDFSDelegationToken(ugi)) {
```

Comment:
This change actually causes the problem: The `ugi` passed into the Kerberos credential check is not necessarily the once for delegation token check.

Specifically in Flink, credentials are checked against the `loginUser` and the delegation token always check against `currentUser` in the case of a delegation or proxy user impersonation. 
See: [HadoopModule.java#L137](https://github.com/apache/flink/blob/master/flink-runtime/src/main/java/org/apache/flink/runtime/security/modules/HadoopModule.java#L137)

I think this is the reason why the CI test fails @tillrohrmann but I can't be sure. @krasinski can you revert this piece of change and try again?